File: 03- Sliding Window/C#/3- Max Consecutive Ones III.cs
public class Solution {
    public int LongestOnes(int[] nums, int k) {
        int l = 0 , r = 0 , zeros = 0 ; 
        int n = nums.Length ; 
        int ans = 0 ; 
        while(r < n){
            if (nums[r] == 0)
                zeros++ ; 
            
            while(l <= r && zeros > k){
                if (nums[l] == 0)
                    zeros-- ; 
                l++ ; 
            }

            ans = Math.Max(ans , r - l + 1);

            r++ ; 
        }

        return ans ; 
    }
}

File: 03- Sliding Window/3- Max Consecutive Ones III.cpp
class Solution {
public:
    int longestOnes(vector<int>& nums, int k) {
        int l = 0 , r = 0 ;
        int zeros = 0 ;
        int ans = 0 ;
        while (r < nums.size()) {
            zeros += (nums[r] == 0) ;

            while (l <= r && zeros > k) {
                zeros -= (nums[l] == 0) ;
                l++ ;
            }

            ans = max(ans , r - l + 1) ;
            r++ ;
        }
        return ans ; 
    }
};